libursa/ZHelpers.cpp

```diff
@@ -1,13 +1,57 @@
 #include "ZHelpers.h"
 
-std::string s_recv(zmq::socket_t &socket) {
+bool s_send_raw(zmq::socket_t *socket, std::string_view payload,
+                int flags = 0) {
+    zmq::message_t message(payload.size());
+    ::memcpy(message.data(), payload.data(), payload.size());
+    return socket->send(message, flags);
+}
+
+bool s_recv_raw(zmq::socket_t *socket, std::string *response) {
     zmq::message_t message;
-    socket.recv(&message);
-    return std::string(static_cast<char *>(message.data()), message.size());
+    bool ok = socket->recv(&message);
+    if (ok) {
+        *response =
+            std::string(static_cast<char *>(message.data()), message.size());
+    }
+    return ok;
+}
+
+void s_send_padding(zmq::socket_t *socket, int flags) {
```

Comment:
case in point: return code ignored :thinking: . Check it and raise runtime error?